tests/integration/utils/flow_tools.py

```diff
@@ -78,22 +84,20 @@ async def _start_flow(
     schd: Scheduler,
     level: int = logging.INFO
 ):
-    """Start a scheduler but don't set it running."""
+    """Start a scheduler but don't set the main loop running."""
     if caplog:
         caplog.set_level(level, CYLC_LOG)
 
-    # install
     await schd.install()
 
-    # start
     try:
         await schd.start()
-        yield caplog
 
-    # stop
     finally:
+        yield caplog
```

Comment:
What's the logic of moving the yield, feels wrong.

Reply:
Yes. Otherwise you won't get the caplog yielded

Reply:
But you have a good point... this now breaks the teardown if an assertion fails in the test

Reply:
Should be addressed now - https://github.com/cylc/cylc-flow/compare/5d7e1d9da1c90f7b50bb0c7ad08adf466014f1cd..04164177e63578bf69578ecf4f36f76e740ffe97